specifyweb/frontend/js_src/lib/components/QueryComboBox/index.tsx

```diff
@@ -508,14 +508,7 @@ export function QueryComboBox({
                 }
               />
             )}
-            {formatted?.resource === undefined ||
-            (hasViewButton &&
-              hasTablePermission(
```

Comment:
no than the viewButton would be there when we don't specify for it in the form def. There is also already a check for the read permission in line 405. So we're good. It's the same logic as for the hasCloneBtn and hasSearchBtn

Reply:
the read check on line 405 has nothing to do with this button doesn't it?
that code path only happens when `isReadOnly`
otherwise, it renders the Pencil button and/or, this viewButton

> no than the viewButton would be there when we don't specify for it in the form def

the view button was being shown when it shouldn't be because of the `formatted?.resource === undefined` condition, which shouldn't have been here - the permission check should have been here, and is not the cause of the bug

I think the code in here should have been changed to this:
```ts
hasViewButton && hasTablePermission(formatted.resource.specifyTable.name, 'create') ? viewButton : undefined
```

Reply:
yes I agree with you, I originally added the `formatted?.resource === undefined` because of: 'formatted.resource' is possibly 'undefined'